app/eq.py

```diff
@@ -128,6 +130,29 @@ async def build(self):
 
         return self._payload
 
+    @staticmethod
+    def encrypt_response_id(questionnaire_id: str):
+        """
+        Encrypts a response_id using the following format: <response_id><first 16 characters of sha256 signature>.
```

Comment:
`<response_id>` should be `<questionnaire_id>` . And we are encrypting the QID to form the response_id so it should be `Creates a response_id by encrypting a questionnaire_id`